src/strategies/Staking.sol

```diff
@@ -4,174 +4,348 @@ pragma solidity =0.8.16;
+        // of maker debt as we pay of compound debt
+        // stEth will be traded to DAI to pay back this flashloan, so we need to set a min amount to make sure the trade succeeds
+        uint daiNeeded;
+        if (makerDaiToPay > compDaiToRedeem && makerDaiToPay - compDaiToRedeem > 0.01e18) {
+            daiNeeded = makerDaiToPay - compDaiToRedeem;
         }
 
-        // WETH to wstETH (via trade)
-        // Uniswap pool is pretty big: https://info.uniswap.org/#/pools/0x840deeef2f115cf50da625f7368c24af6fe74410
-        ISwapRouter.ExactInputSingleParams memory params = ISwapRouter.ExactInputSingleParams({
-            tokenIn: address(WETH),
-            tokenOut: address(WSTETH),
-            fee: 500,
-            recipient: address(this),
-            deadline: block.timestamp,
-            amountIn: amount,
-            amountOutMinimum: 0,
-            sqrtPriceLimitX96: 0
-        });
-        ROUTER.exactInputSingle(params);
-
-        // Supply wstEth on AAVE
-        AAVE.supply({
-            asset: address(WSTETH),
-            amount: WSTETH.balanceOf(address(this)),
-            onBehalfOf: address(this),
-            referralCode: 0
-        });
-
-        // Borrow Eth. TODO: consider converting WSTETH to ETH using oracle prices
-        // TODO: make sure collat ratio is safe
-        AAVE.setUserEMode(1); // Allow up to 90% borrow
-        AAVE.borrow({
-            asset: address(WETH),
-            amount: amount - posSize,
-            interestRateMode: 2,
-            referralCode: 0,
-            onBehalfOf: address(this)
-        });
+        // Flashloan `ethNeeded` eth from balancer, _endPosition gets called
+        uint arrSize = daiNeeded > 0 ? 2 : 1;
+        ERC20[] memory tokens = new ERC20[](arrSize);
+        if (daiNeeded > 0) {
+            tokens[0] = DAI;
+            tokens[1] = WETH;
+        } else {
+            tokens[0] = WETH;
+        }
+        uint256[] memory amounts = new uint256[](arrSize);
+        if (daiNeeded > 0) {
+            amounts[0] = daiNeeded;
+            amounts[1] = ethNeeded;
+        } else {
+            amounts[0] = ethNeeded;
+        }
+        console.log("daiNeeded", daiNeeded);
+
+        balancer.flashLoan({recipient: IFlashLoanRecipient(address(this)), tokens: tokens, amounts: amounts, userData: abi.encode(Loan.close)});
+
+        // Unlocked collateral is equal to my current weth balance
+        // Send eth back to user
+        uint wethToSend = WETH.balanceOf(address(this));
+        WETH.safeTransfer(msg.sender, wethToSend);
+        return wethToSend;
+     }
+
+       function _endPosition(uint256 ethBorrowed, uint daiBorrowed) internal {
+        // Pay debt in compound
+        uint ethDebt = cETH.borrowBalanceCurrent(address(this));
+        IWETH(address(WETH)).withdraw(ethBorrowed);
+
+        cETH.repayBorrow{value: ethBorrowed}();
+        uint daiToRedeem = cDAI.balanceOfUnderlying(address(this)).mulDivDown(ethBorrowed, ethDebt);
```

Comment:
Should we keep 75%, will require less rebalance.

Reply:
See the strategy doc.